File: projects/08/java/TempTester.java
/* NameTester.java is a 'driver' to test class Name.
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Student: Sinai Park(sp46)
 * Date: April 10 2020
 ************************************************************/


 import java.io.*;
 import java.util.Scanner;

public class TempTester  
{
  public static void main(String[] args)
   {
      Scanner keyboard = new Scanner(System.in);
      //set starting temp
      Temp start_temp = new Temp(35.6, 'F');

      // Prompt the user for the limit temp value
      System.out.println("Enter the start temp value:");
      start_temp.readTemp(keyboard);  
      //start_temp.printTemp();

      Temp limit_temp = new Temp(35.6, 'F');
      
	// Prompt the user for the limit temp value
      System.out.println("Enter the limit temp value:");
      limit_temp.readTemp(keyboard);

      // Prompt the user for the increment value
	System.out.println("Please enter the increment value: ");
	double increment = keyboard.nextDouble();

	char starting_scale = start_temp.getScale();


	// While loop to print out the temperature values in a table format in each scale
	while ( start_temp.less_than(limit_temp ) )
	{
	//in the first column print out the Fahrenheit degrees
	start_temp.toFarenheit();
	start_temp.printTemp();
	System.out.print("\t");
	
	//in the second column print out the Celsius degrees
	start_temp.toCelcius();
	start_temp.printTemp();
	System.out.print("\t");

	//in the third column print out the Kelvin degrees
	start_temp.toKelvin();
	start_temp.printTemp();
	System.out.print("\n");

	// return the starting scale back to that Scale to add inrement
	if ( starting_scale == 'F' || starting_scale == 'f' )
	{
		start_temp.toFarenheit();
	}

	if ( starting_scale == 'C' || starting_scale == 'c' )
	{
		start_temp.toCelcius();
	}

	if ( starting_scale == 'K' || starting_scale == 'k' )
	{
		start_temp.toKelvin();
	}

	// Increment the start_temp
	start_temp.raise(increment);
	}     

      //System.out.println("All tests passed!");
  }
}

File: projects/08/clojure/src/tempTester.clj
;;;; tempTester.clj tests a clojure Temp type.
;;;;
;;;; Output: results of testing Temp functions.
;;;;
;;;; Usage: clojure -m nameTester
;;;;
;;;; Begun by: Dr. Adams, CS 214 at Calvin College.
;;;; Completed by: Sinai Park (sp46)
;;;; Date: April 1 2020
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns tempTester)   ; program name

(load "Temp")


;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; A simple driver to test our Temp functions. 
;;; Output: the result of testing our Temp functions.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
  (let
    [                                          ; 3 ways to construct an object:
       start_temp (make-Temp 30.0 "F")  ; -using our "make-" constructor
       limit_temp (->Temp 30.0 "F" ) ; -invoking constructor directly                                                   
       name3 (map->Temp {:degrees 30.0 :scale "Jinx"}) ; -mapping field-names to values; 
    ]
    (printTemp start_temp) (println)
    (println "Starting Temperature ")
    (let 
      [start_temp (readTemp start_temp)]
    (println "Limiting Temperature ") 
    (let 
      [limit_temp (readTemp limit_temp)]
    (println "Enter the increment number: ") 
    (let 
      [increment (read)]

   
      ;First print out the Fahrenheit degrees
      (toFahrenheit start_temp)
      (printTemp start_temp)
      (print "\t")
      (raise start_temp increment)
  
      ;Second print out the Celsius degrees
      (toCelsius start_temp)
      (printTemp start_temp)
      (print "\t")
  
      ;Third print out the Kelvin degrees
      (toKelvin start_temp)
      (printTemp start_temp)
      (println "")

    ))) ;three let definitions
    (println "\nAll tests passed!")

  )
)

